### PR TITLE
Refactor Bossphorus

### PR DESCRIPTION
**Pull request by mlw214**
@j6k4m8 Lots of changes, let me know what you think. I reran the benchmarks to make sure everything still works.

Once you're OK with these changes, I'm going to start working on the proxy/caching features.

### CODE REVIEW

**Review by j6k4m8 — DISMISSED** [review]
Nice one, re: moving app constructor logic

File: bossphorus/storagemanager/_FilesystemStorageManager.py [hunk 1]
```diff
@@ -18,51 +17,29 @@
...
 
     Contains logic for reading and writing to local filesystem.
     """
 
-    def __init__(
-            self,
-            upload_path: str = UPLOADS_PATH,
-            block_size: [int, int, int] = BLOCK_SIZE
-        ):
+    def __init__(self, storage_path: str, block_size: [int, int, int]):
```
Comment (j6k4m8) [inline bossphorus/storagemanager/_FilesystemStorageManager.py:30]: Rationale for a centralized default block size: 

It's important to keep block-size consistent between runs, or the filesystem manager won't be able to find the files on disk. As long as we enforce staying robust to this (or guess the block-size from existing files in the filesystem) maybe this doesn't matter?

Comment (mlw214) [reply bossphorus/storagemanager/_FilesystemStorageManager.py:30]: That's a good point. I have a suggestion that I think will alleviate your concern, but also let it be configurable: when bossphorus launches with a given storage directory, it checks the root of the storage directory for a config file (probably a hidden dotfile) for the block size. If it exists, it loads the blocksize from that file and overrides whatever was passed at configuration. If it doesn't exist, it uses the configured blocksize, and saves that blocksize to a dotfile in the root storage directory.

Thoughts?

**Comment by j6k4m8** [comment]
I like!

**Review by j6k4m8 — APPROVED** [review]
Realized I nulled this review by committing. Approving again!

**Pull request merged**